src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -159,7 +159,8 @@ public static bool IsImplementationOfInterfaceMethod(this IMethodSymbol method,
         public static bool IsDisposeImplementation(this IMethodSymbol method, Compilation compilation)
```

Comment:
This helper method is used in a lot of places and I am not sure if this is the right idea to update it. For example, this means that the dispose pattern implementation will now raise issues when you implement `DisposeAsync` which is probably not what we want. IMO we should either create a new helper, use [GetDisposeMethodKind](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs#L280) or call [method.ContainingType.IsDisposable(iDisposable, iAsyncDisposable)](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs#L287) @mavasani WDYT?

Reply:
Creating a new helper (`IsAsyncDisposeImplementation`) sounds good. But let me wait for @mavasani

Reply:
Thanks, yes a new helper sounds good. Can you also audit the existing callers of `IsDisposeImplementation` to verify they are handling IAsyncDisposable where appropriate?

Reply:
I only found [this](https://github.com/dotnet/roslyn-analyzers/blob/75d6b9c80862070fc1b5efd0b41bd750152e3c0f/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/DoNotCallDangerousMethodsInDeserialization.cs#L214) that calls `IsDisposeImplementation`, but I don't know if it should be handling AsyncDispose or not.